app/connectors/LibraryConnector.scala

```diff
@@ -0,0 +1,25 @@
+package connectors
+
+import models.APIError
+import play.api.libs.json.OFormat
+import cats.data.EitherT
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+import play.api.libs.ws.{WSClient, WSResponse}
+
+class LibraryConnector @Inject()(ws: WSClient) {
+  def get[Response](url: String)(implicit rds: OFormat[Response], ec: ExecutionContext): EitherT[Future, APIError, Response] = {
+    val request = ws.url(url)
+    val response = request.get()
+    EitherT {
+      response
+        .map {
+          result =>
+            Right(result.json.as[Response])
```

Comment:
Probably not worth doing it for this project since I can see that this is mostly copied in from the instructions - but something to keep in mind for the next project's connector is handling if the `.as[Response]` fails. It looks like it throws an exception if it fails so maybe a try-catch would work. Maybe a match but not too sure.